### DISCUSSION TITLE
queryClient.fetchQuery doesn't deduplicate requests

### DISCUSSION THREAD

**Discussion by vitaliidm** [TanStack/query#5638] [Q&A]
Hey community!

I have encountered issue using `queryClient.fetchQuery`.

I have a test function 

```ts

      return queryClient.fetchQuery({
        queryKey: [myKey],
        queryFn: async () => fetchData(myKey),
        staleTime: 60 * 1000,
      });

```

It called from 2 places in my react app, when myKey is changed from input.
I would expect only one request to be made, since `myKey` is the same in both invocations of `queryClient.fetchQuery` and I defined `staleTime`. But, instead I can see 2 separate requests, with the same `myKey`. Moreover, when I update `myKey` to old value, it doesn't returned form cache, instead I see another 2 requests.
Can anyone advice why it happens?

My version `4.29.12`
Thanks

**Comment by TkDodo** [comment]
show this behavior in a reproduction codesandbox please
